src/V3AstNodeDType.h

```diff
@@ -61,9 +61,9 @@ class AstNodeDType VL_NOT_FINAL : public AstNode {
     /// under another compound-requiring object e.g. class
     virtual bool isCompound() const = 0;
     // (Slow) recurse down to find basic data type
-    virtual AstBasicDType* basicp() const = 0;
+    virtual AstBasicDType* basicp() const VL_MT_SAFE = 0;
```

Comment:
What did we decide on earlier pull related to these: This is safe only if assuming some other visitor isn't changing the tree topology. I think we need e.g. VL_MT_STABLE_TREE or something to state that assumption

Reply:
Sure, I can add ``VL_MT_STABLE_TREE``, but would it mean, that every ``basicp`` will need to request exclusive access to be executed in MT_SAFE context or we only want to state this assumption? I would like to avoid requesting exclusive access is this function as it is used often.

Reply:
By VL_MT_STABLE_TREE I mean the assumption is it does not do locking and the user can only call it when they know they are not changing the tree in another thread.  I think this use matches what you are saying you want to use it for.

I'm thinking we may end up with algorithms that DO modify the tree in parallel (e.g. many algorithms can operate on different modules in parallel), so it's good to know which are legal to use.